packages/transaction-controller/CHANGELOG.md

```diff
@@ -7,6 +7,37 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+- `EtherscanRemoteTransactionSource` now enforces a 5 second delay between requests to avoid rate limiting ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `TransactionMeta` type now specifies an optional `networkClientId` field ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `startIncomingTransactionPolling()` now accepts an optional array of `networkClientIds`. When provided, the `IncomingTransactionHelpers` for those networkClientIds will be started. If empty or not provided, the global `IncomingTransactionHelper` will be started. ([#3643](https://github.com/MetaMask/core/pull/3643))
+- `stopIncomingTransactionPolling()` now accepts an optional array of `networkClientIds`. When provided, the `IncomingTransactionHelpers` for those networkClientIds will be stoppped. If empty or not provided, the global `IncomingTransactionHelper` will be stopped. ([#3643](https://github.com/MetaMask/core/pull/3643))
+
+### Added
+
+- **BREAKING:** Constructor now expects a `getNetworkClientRegistry` callback function ([#3643](https://github.com/MetaMask/core/pull/3643))
+- **BREAKING:** Messenger now requires `NetworkController:stateChange` to be an allowed event ([#3643](https://github.com/MetaMask/core/pull/3643))
+- **BREAKING:** Messenger now requires `NetworkController:findNetworkClientByChainId` and `NetworkController:getNetworkClientById` actions ([#3643](https://github.com/MetaMask/core/pull/3643))
+- Adds a feature flag parameter `isMultichainEnabled`  passed via the constructor (and defaulted to false), which when passed a truthy value will initialize `IncomingTransactionHelper`, `PendingTransactionTracker`, `NonceTracker` per networkClientId from the NetworkController’s `networkClientRegistry` and a EtherScanRemoteTransactionSource helper per chainId in the registry. ([#3643](https://github.com/MetaMask/core/pull/3643))
+- Adds `destroy()` method that stops/removes internal polling and listeners ([#3643](https://github.com/MetaMask/core/pull/3643))
+- Adds `stopAllIncomingTransactionPolling()` method that stops the global `IncomingTransactionHelper` and each networkClientId's `IncomingTransactionHelper` ([#3643](https://github.com/MetaMask/core/pull/3643))
+- Exports `PendingTransactionOptions` type ([#3643](https://github.com/MetaMask/core/pull/3643))
+- Exports `TransactionControllerOptions` type ([#3643](https://github.com/MetaMask/core/pull/3643))
+
+### Fixed
+
+- Approval of transactions previously used the chainId from the global provider state as the source of truth. Instead it will now use the chainId from the TransactionMeta object. **Not sure we really need to put this in the changelog since all tx will have chainId populated** ???? ([#3643](https://github.com/MetaMask/core/pull/3643))
```

Comment:
is this needed? I guess it doesn't hurt to leave

Reply:
I think we can remove this

Reply:
done https://github.com/MetaMask/core/pull/3925/commits/067e31be638d2473016127161df669de5d925804